integrations/operator/sidecar/tbot.go

```diff
@@ -139,6 +138,42 @@ func (b *Bot) GetClient(ctx context.Context) (*client.Client, error) {
 	return c, trace.Wrap(err)
 }
 
+// GetClient gets a client authenticated as the bot. To avoid rebuilding a client for each call, this function caches
+// the client and creates a new one only when the client certs changed (tbot renewed them).
+func (b *Bot) GetClient(ctx context.Context) (*client.Client, error) {
+	b.mutex.Lock()
+	defer b.mutex.Unlock()
+
+	if !b.running {
+		return nil, trace.Errorf("bot not started yet")
+	}
+	// If the bot has not joined the cluster yet or not generated client certs we bail out
+	// This is either temporary or the bot is dead and the manager will shut down everything.
+	storageDestination := b.cfg.Storage.Destination
+	if botCert, err := storageDestination.Read(ctx, identity.TLSCertKey); err != nil || len(botCert) == 0 {
+		return nil, trace.Retry(err, "bot cert not yet present")
+	}
+
+	cert, err := b.cfg.Outputs[0].GetDestination().Read(ctx, identity.TLSCertKey)
+	if err != nil || len(cert) == 0 {
+		return nil, trace.Retry(err, "cert not yet present")
+	}
+
+	// This is where caching happens. We don't know when tbot renews the certificates, so we need to check
+	// if the current certificate stored in memory changed since last time. If it did not and we already built a
+	// working client, then we hit the cache. Else we build a new client and update the
+	if b.cachedClient != nil && bytes.Equal(cert, b.cachedCert) {
+		return b.cachedClient, nil
+	}
+
+	freshClient, err := b.clientBuilder(ctx)
```

Comment:
are we ever closing the client?

Reply:
This is the leak I spotted - the problem also exists in the existing code. We can probably adjust this PR to fix both at once :')

Reply:
Fixed in 13de098a38a33c7890cc95bd6886c6a41fa8e4aa

This is not super clean, and we'll definitely want to get rid of this when tbot will send us clients with in-place cert-renewal. However, tbot changes won't be backported to v12/v13, so we need the current fix for those versions.